tempesta_fw/connection.h

```diff
@@ -177,7 +177,9 @@ tfw_connection_put(TfwConnection *conn)
 static inline void
 tfw_connection_put_to_death(TfwConnection *conn)
 {
-	atomic_add(TFW_CONN_DEATHCNT, &conn->refcnt);
+	/* Do not put dead connections to death state */
+	if (tfw_connection_nfo(conn))
```

Comment:
What about `refcnt = 0`? It seems it's invalid case, so it has sense to add BUG_ON() for it.